src/Autofac.Integration.Wcf/AutofacInstanceContext.cs

```diff
@@ -103,7 +105,17 @@ public AutofacInstanceContext(ILifetimeScope container)
             {
                 throw new ArgumentNullException("container");
             }
-            this.OperationLifetime = container.BeginLifetimeScope();
+            this.OperationLifetime = container.BeginLifetimeScope("WcfRequest", (builder) =>
+            {
+                var jitModules = container.ResolveOptional<IPerInstanceContextJitModuleContainer>();
```

Comment:
This is a potential hit for every request though this isn't the "80% use case." Unfortunately we get hit with complaints about performance at every turn. Have you looked at how this impacts performance, e.g., with some benchmarks before/after?

I'm curious if we may need to, instead, add some sort of "enable this feature" flag somehow so you have to turn it on before it goes through this process. We do that for things like attribute filtering on objects because that's an unnecessary hit if folks don't use it.

Of course, if the hit isn't that bad, maybe it's not needed.